Add ABI bufr2IODA python converter and JSON file.

azadeh-gh opened the pull request:
[#115](https://github.com/NOAA-EMC/JEDI-T2O/issues/115)
**New files include:**
- `parm/ioda/bufr2ioda/bufr2ioda_gsrcsr.json`  : JSON containing data format, sensor/bufr2ioda_gsrcsr.json, and satellite information
-  `ush/ioda/bufr2ioda/bufr2ioda_gsrcsr.py` : bufr2ioda python code for converting ABI GOES-16, GOES-17, and GOES-18 from BUFR to IODA

Commit:
adding python and json file

Commit:
adding python and json file

Commit:
only for testinggit add parm/atm/jcb-base.yaml.j2 parm/atm/jcb-prototype_3dvar.yaml.j2git add parm/atm/jcb-base.yaml.j2 parm/atm/jcb-prototype_3dvar.yaml.j2!

Commit:
only for testinggit add parm/atm/jcb-base.yaml.j2 parm/atm/jcb-prototype_3dvar.yaml.j2git add parm/atm/jcb-base.yaml.j2 parm/atm/jcb-prototype_3dvar.yaml.j2!

Commit:
modified python converter and json

Commit:
modified python converter and json

Commit:
modified sensorScanPosition

Commit:
modified sensorScanPosition

Commit:
ABI bufr2IODA python converter

Commit:
ABI bufr2IODA python converter

Commit:
updated jcb-base.yaml.j2

Commit:
updated jcb-base.yaml.j2

Commit:
python norms

Commit:
python norms

Commit:
code norms

Commit:
code norms

Commit:
code norms

Commit:
code norms

Commit:
code norms

Commit:
code norms

CoryMartin-NOAA committed:
add ensemble mean capability (#1335)

Add ensmeanandvariance to gdas.x

CoryMartin-NOAA commented on parm/atm/jcb-prototype_3dvar.yaml.j2:
```diff
@@ -29,3 +29,11 @@ observations:
+#  - ascatw.ascat_metop-b
+  #- conv_ps  # Turned off until naming convention of obs files is resolved.
+  #- gnssro   # Turned off until naming convention of obs files is resolved.
+#  - ompsnp_npp
```
why are all these others added?

apchoiCMD committed:
Add ABI sea-ice product to ioda converter (#1280)

#### Description
- Task for adding a new ioda converter for new `ABI` type sea-ice
concentration
- For sea-ice obs, AMSR2, MIRS, JPSSRR and ABI products will be used
- For the part of conversion of metadata, the reference is
[here](https://www.goes-r.gov/products/docs/PUG-L2+-vol5.pdf)

Close #1182

---------

Co-authored-by: Guillaume Vernieres <guillaume.vernieres@noaa.gov>

givelberg committed:
fixing the yaml (#1338)

corrected the yaml for tropical moorings

guillaumevernieres committed:
Addition of a gnu lua file for hercules (#1341)

- fixes #1340 

I had to comment out `da-utils` in the bundle, but other than that the
system builds.

CoryMartin-NOAA committed:
fix CI for aerosol DA (#1344)

@RussTreadon-NOAA @KateFriedman-NOAA I think including this in the
GDASApp hash in G-W should fix the aero CI (hopefully)

DavidNew-NOAA committed:
Fix jjob testing in related to change if setup_expt.py (#1348)

Resolves GDASApp issue
[#1347](https://github.com/NOAA-EMC/GDASApp/issues/1347)

jjob tests pass with this change now, but you have to revert
```test/atm/global-workflow/config.yaml``` to
https://github.com/NOAA-EMC/GDASApp/commit/6fb0a655ffe61c6dd4f2acaa4c4490121ba980fb
since jjob testing will not work without modification until Global
Workflow PR
[#2992](https://github.com/NOAA-EMC/global-workflow/pull/2992) is
merged.

Commit:
Merge branch 'feature/abi_validation' of https://github.com/NOAA-EMC/GDASApp into feature/abi_validation

Commit:
added abi

CoryMartin-NOAA committed:
one line

CoryMartin-NOAA committed:
revert one line
